src/providerAsMiddleware.ts

```diff
@@ -10,19 +10,15 @@ export function providerAsMiddleware(
   provider: SafeEventEmitterProvider,
 ): JsonRpcMiddleware<JsonRpcParams, Json> {
   return (req, res, _next, end) => {
```

Comment:
Hmm, I can't add a suggestion here for this, but can we simplify this to use `createAsyncMiddleware`?

``` typescript
  return createAsyncMiddleware(async (req, res, _next) => {
    res.result = await provider.request(req);
  });
```